### PR TITLE
feat(compass-e2e-tests): Add basic end-to-end tests COMPASS-4992

### PR DESCRIPTION
**Pull request by gribnoysup**
## Description

This PR adds initial setup for e2e tests + a few very basic integration tests on the hottest path in the application: connection flow. To make this whole thing work I had to additionally change a bunch of other stuff across the codebase, here's the gist of those changes:

- **hadorn-build** got it release script updated in a way that allows to use some release tasks programmaticaly. Reason for that change is that hadron compile cache was breaking spectron runtime in a weird way, as this whole cache code is on it's way out instead of digging too deep I decided to just precompile hadron assets before runing the tests, it's a quick process and allows to run these tests against your local changes in an environment as close to production as possible.

- **compass-connect** and **compass-shell** got a bunch of `data-test-id`s added and as a drive-by fix I did a few semantic fixes replacing links with buttons where appropriate, switching to leafygreen, and fixing some aria attributes

## Motivation and Context
We want to have e2e tests in CI to have more confidence that all out changes still keep Compass in the usable state
- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

## TODO

As I haven't yet tried this in CI, there are a few things to make sure before we merge it:

- [ ] Make sure GitHub Actions can run those tests and the tests are not flaky
- [ ] Make sure Evergree can run those tests and the tests are not flaky

### CODE REVIEW

**Review by rose-m — APPROVED** [review]
Don't understand the Hardon stuff in detail but that's an amazing PR!! Love it 👏

File: packages/hadron-build/commands/release.js [hunk 1]
```diff
@@ -164,27 +166,6 @@ const createBrandedApplication = (CONFIG, done) => {
   });
 };
 
-/**
- * Create the Windows NSIS installer.
- *
- * @param {Object} CONFIG - The config.
- * @param {Function} done - The callback.
- */
-const createNsisApplication = (CONFIG, done) => {
```
Comment (gribnoysup) [inline packages/hadron-build/commands/release.js:173]: This was not used anywhere in the build process

**Comment by gribnoysup** [comment]
This is now in sync with main and here's an [evergreen patch](https://spruce.mongodb.com/version/611e1f3a0305b914998bb15b/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) to confirm it's working there

**Review by rose-m — COMMENTED** [review]
Confused bit: the `packages/compass/src/app/index.html` shows 11.000 lines changed :D I'm not sure that's intended?^^

File: .evergreen.yml [hunk 2]
```diff
@@ -375,57 +384,57 @@ tasks:
 
       - func: install
 
-      - func: verify
+      # - func: verify
```
[the remark this answers is not in the document]
Comment (gribnoysup) [reply .evergreen.yml:387]: This and the html you mentioned were commited by mistake, thanks for catching it! 🙏

**Review by rose-m — APPROVED** [review]

**Pull request merged**